client/src/main/java/org/kakara/client/scenes/MainGameScene.java

```diff
@@ -1,20 +1,143 @@
+            Material mt = new Material(TextureCache.getInstance(resourceManager).getTexture(resource.getLocalPath(), this));
+            mesh.setMaterial(mt);
+            MeshGameItem gi = new MeshGameItem(mesh);
+            gi.setPosition(MoreUtils.locationToVector3(gameBlock.getLocation()));
+            gi.setCollider(new ObjectBoxCollider(false, true));
+
+            vector3f = MoreUtils.locationToVector3(gameBlock.getLocation());
+            add(gi);
+
+        }
+
+        light = new PointLight(new Vector3f(0, 2, 0));
+        //lightIndication = new MeshGameItem(mesh).setScale(0.3f).setPosition(0, 2, 0);
+        add(light);
+        this.add(new PointLight().setPosition(12, 12, 0).setDiffuse(0.1f, 0, 0).setSpecular(1.0f, 0, 0));
+        this.add(new PointLight().setPosition(3, 3, 3).setDiffuse(0f, 0.3f, 0).setSpecular(0, 0, 0.7f));
+        this.add(new SpotLight(gameHandler.getCamera().getPosition(), new Vector3(0, 0, 1)));
+        // this.add(lightIndication);
+        // Allows you to see the light.
+        this.getLightHandler().getDirectionalLight().setDirection(12, 12, 0);
+
+
+        //Load Player
+
+
+        Mesh[] mainPlayer = StaticModelLoader.load(resourceManager.getResource("player/steve.obj"), "/player", this, resourceManager);
+        player = new MeshGameItem(mainPlayer);
+        player.setPosition(12,12,12);
+        player.setScale(0.3f);
+        player.setCollider(new BoxCollider(new Vector3(0, 0, 0), new Vector3(1, 1.5f, 1)));
+        player.getCollider().setUseGravity(true).setTrigger(false);
+        ((BoxCollider) player.getCollider()).setOffset(new Vector3(0, 0.7f, 0));
+        add(player);
+    }
+
+    public void loadMods() {
+        kakaraGame.getKakaraCore().getModManager().loadMods(Arrays.stream(getModJars()).collect(Collectors.toList()));
+    }
+
+    private File[] getModJars() {
+        File dir = new File("test" + File.separator + "mods");
+
+        return dir.listFiles(new FilenameFilter() {
+            public boolean accept(File dir, String filename) {
+                return filename.endsWith(".jar");
+            }
+        });
     }
 
     @Override
     public void update() {
+        if (player == null) return;
 
+        KeyInput ki = kakaraGame.getGameHandler().getKeyInput();
+Vector3 oldP = player.getPosition();
+        if (ki.isKeyPressed(GLFW_KEY_W)) {
+            player.movePosition(0, 0, -1);
+        }
+        if (ki.isKeyPressed(GLFW_KEY_S)) {
+            player.movePosition(0, 0, 1);
+        }
+        if (ki.isKeyPressed(GLFW_KEY_A)) {
+            player.movePosition(-1, 0, 0);
+        }
+        if (ki.isKeyPressed(GLFW_KEY_D)) {
+            player.movePosition(1, 0, 0);
+        }
+        if (ki.isKeyPressed(GLFW_KEY_SPACE)) {
+            player.movePosition(0, 1, 0);
```

Comment:
Not the best way to handle jumping, but it does work.

Reply:
I know.